dynatrace_health/stackstate_checks/dynatrace_health/dynatrace_health.py

```
@@ -203,21 +208,21 @@ def _collect_events(self, dynatrace_client, instance_info):
             event_limit_reached = str(e)
         return new_events, event_limit_reached
 
-    def _get_events(self, dynatrace_client, url, from_time=None, cursor=None):
+    def _get_events(self, dynatrace_client, url, from_time=None, next_page_key=None):
         """
         Get events from Dynatrace Event API endpoint
         :param dynatrace_client: dynatrace rest client
         :param url: dynatrace instance url
         :param from_time: timestamp from which to collect events
-        :param cursor: batch cursor
+        :param next_page_key: batch cursor
         :return: Event API endpoint response
         """
         params = {}
         if from_time:
             params['from'] = from_time
-        if cursor:
-            params['cursor'] = cursor
-        endpoint = dynatrace_client.get_endpoint(url, "/api/v1/events")
+        if next_page_key:
+            params['nextPageKey'] = next_page_key
+        endpoint = dynatrace_client.get_endpoint(url, "/api/v2/events")
         events = dynatrace_client.get_dynatrace_json_response(endpoint, params)
```

how are errors communicated here? does this throw on non-200 responses such that we fail fast/hard?

---

I think the line you should be concerned about is line 226... and non-200 responses are dealt with inside `get_dynatrace_json_response` as follows:
```
                if response.status_code != 200:
                    if "error" in response_json:
                        msg = response_json["error"].get("message")
                    else:
                        msg = "Got %s when hitting %s" % (response.status_code, endpoint)
                    self.log.error(msg)
                    raise Exception(
                        'Got an unexpected error with status code %s and message: %s' % (response.status_code, msg))
```